src/mdm-term.resource.ts

```diff
@@ -39,79 +40,133 @@ import { IMdmQueryStringParams, IMdmRestHandlerOptions } from './mdm-rest-handle
-    /// <param name="termId">Terminology Id or a path in the format typePrefix:label|typePrefix:label</param>
-    /// <param name="queryStringParams">Query String Params</param>
-    /// <param name="restHandlerOptions">restHandler Options</param>
-    get(terminologyId: string, termId: string, queryStringParams?: IMdmQueryStringParams, restHandlerOptions?: IMdmRestHandlerOptions) {
-        let url = '';
-        if (this.isGuid(terminologyId)) {
-            url = `${this.apiEndpoint}/terminologies/${terminologyId}/terms/${termId}`;
-        }
-        else {
-            url = `${this.apiEndpoint}/terminologies/path/${termId}`;
-        }
-
-        return this.simpleGet(url, queryStringParams, restHandlerOptions);
-    }
 
-    addTermRelationships(terminologyId: string, termId: string, data: any, restHandlerOptions?: IMdmRestHandlerOptions) {
-        const url = `${this.apiEndpoint}/terminologies/${terminologyId}/terms/${termId}/termRelationships`;
-        return this.simplePost(url, data, restHandlerOptions);
-    }
-
-    termRelationships(terminologyId: string, termId: string, queryStringParams?: IMdmQueryStringParams, restHandlerOptions?: IMdmRestHandlerOptions) {
-        const url = `${this.apiEndpoint}/terminologies/${terminologyId}/terms/${termId}/termRelationships`;
-        return this.simpleGet(url, queryStringParams, restHandlerOptions);
-    }
+/**
+ * MDM resource for the management of terms within terminologies.
+ *
+ * @see {@link MdmTerminologyResource}
+ */
+export class MdmTermResource extends MdmResource {
 
-    removeTermRelationship(terminologyId: string, termId: string, termRelationshipId: string, queryStringParams?: IMdmQueryStringParams, restHandlerOptions?: IMdmRestHandlerOptions) {
-        const url = `${this.apiEndpoint}/terminologies/${terminologyId}/terms/${termId}/termRelationships/${termRelationshipId}`;
-        return this.simpleDelete(url, queryStringParams, restHandlerOptions);
+  search(terminologyId: string, queryStringParams?: QueryParameters, restHandlerOptions?: RequestSettings) {
+    const url = `${this.apiEndpoint}/terminologies/${terminologyId}/terms/search`;
+    return this.simplePost(url, queryStringParams, restHandlerOptions);
+  }
+
+  tree(terminologyId: string, termId?: string, queryStringParams?: QueryParameters, restHandlerOptions?: RequestSettings) {
+    const url = `${this.apiEndpoint}/terminologies/${terminologyId}/terms/tree${termId ? `/${termId}` : ''}`;
+    return this.simpleGet(url, queryStringParams, restHandlerOptions);
+  }
+
+  /**
+   * `HTTP POST` - Creates a new term under a chosen terminology.
+   *
+   * @param terminologyId The unique identifier of the terminology to create the term under.
+   * @param data The payload of the request containing all the details for the term to create.
+   * @param query Optional query parameters to control the creation of the term, if required.
+   * @param options Optional REST handler parameters, if required.
```

Comment:
mentions query and options on the params list there are none in the actual method

Reply:
Fixed